ilastik/applets/pixelClassification/pixelClassificationSerializer.py

```diff
@@ -29,12 +32,81 @@
 
 
+class BackwardsCompatibleLabelSerialBlockSlot(SerialBlockSlot):
+    def get_corresponding_input_image_slot(self, labelSlot: OutputSlot) -> OutputSlot:
+        labelMultislot = labelSlot.operator.inputs[labelSlot.name]
+        for subslotIndex in range(len(labelMultislot)):
+            if labelMultislot[subslotIndex] == labelSlot:
+                return labelSlot.operator.InputImages[subslotIndex]
+
+    def get_input_image_original_axiskeys(self, labelSlot: OutputSlot) -> str:
+        return "".join(self.get_corresponding_input_image_slot(labelSlot).meta.getOriginalAxisKeys())
+
+    def get_input_image_current_axiskeys(self, labelSlot: OutputSlot) -> str:
+        return "".join(self.get_corresponding_input_image_slot(labelSlot).meta.getAxisKeys())
+
+    def deserialization_requires_data_conversion(self, project) -> bool:
+        return self.labels_were_saved_in_forced_canonical_order(project)
+
+    def labels_were_saved_in_forced_canonical_order(self, project: Project) -> bool:
+        pixel_plus_object_workflow_name = "Object Classification (from pixel classification)"
+        v1_3_3 = parse_version("1.3.3")
+        v1_3_3post2 = parse_version("1.3.3post2")
+
+        return (
+            v1_3_3 <= project.ilastikVersion < v1_3_3post2 and project.workflowName == pixel_plus_object_workflow_name
+        )
+
+    def get_saved_data_axiskeys(self, slot: OutputSlot, project: Project) -> str:
+        if self.labels_were_saved_in_forced_canonical_order(project):
+            return "txyzc"
+        else:
+            return self.get_input_image_original_axiskeys(slot)
+
+    def reshape_datablock_and_slicing_for_input(
+        self, block: numpy.ndarray, slicing: List[slice], slot: OutputSlot, project: Project
+    ) -> Tuple[numpy.ndarray, List[slice]]:
+        """Reshapes a block of data and its corresponding slicing into the slot's current shape, so as to be
+        compatible with versions of ilastik that saved and loaded block slots in their original shape
+
+        Checks for version 1.3.3 and 1.3.3post1 because those were the versions that saved labels in 5D
+        """
+        current_axiskeys = self.get_input_image_current_axiskeys(slot)
+        saved_data_axiskeys = self.get_saved_data_axiskeys(slot, project)
+        fixed_slicing = Slice5D.zero(**dict(zip(saved_data_axiskeys, slicing))).to_slices(current_axiskeys)
+        fixed_block = Array5D(block, saved_data_axiskeys).raw(current_axiskeys)
+        return fixed_block, fixed_slicing
+
+    def reshape_datablock_and_slicing_for_output(
+        self, block: numpy.ndarray, slicing: List[slice], slot: OutputSlot
+    ) -> Tuple[numpy.ndarray, List[slice]]:
+        """Reshapes a block of data and its corresponding slicing into the slot's original shape, so as to be
+        compatible with versions of ilastik that saved and loaded block slots in their original shape
+
+        Always save using original shape to be backwards compatible with 1.3.2
```

Comment:
So i think version `1.3.2` is _forwards_ compatible with `1.3.3`, i.e., it produces files that can be interpreted by future versions. Since this PR upgrades ilastik to `1.3.3post2`, and it intends to make `1.3.3post2` compatible with versions from the past, I'd say it's trying to achieve backwards compatibility

Reply:
yeah sorry, my bad :)